org.eclipse.acute/src/org/eclipse/acute/dotnetnew/DotnetNewWizard.java

```diff
@@ -34,8 +35,16 @@
 	private DotnetNewWizardPage wizardPage;
 	@Override
 	public void init(IWorkbench workbench, IStructuredSelection selection) {
-		// TODO
 		wizardPage = new DotnetNewWizardPage();
+
+		Object firstElement = selection.getFirstElement();
+		if (firstElement instanceof IResource) {
```

Comment:
If there are a lot of working sets and projects, it may take a while to search for containing working sets. The options I see are to:

1. Wait for the list of working sets to be made before building the wizard (makes the UI wait for loading, so not great)

2. Load the wizard but restrict the working sets section and say 'loading' until the list is loaded and populates the area (makes the user wait if they wish to change the working set section)

3. Load everything, let the user edit the working sets section then:
3a. Add the selected file's working sets to the user added working sets (may result in the user not noticing the change and finishing the wizard with adding the new project to unwanted working sets)
3b. Override the users selection with the selected file's working sets (Overrides the users input)

Which is the best option?